dev.c

```diff
@@ -20,67 +20,89 @@
 #include <pci.h>
 #include <dev.h>
 
-u32 dev_locate(void)
+/*
+ * There are only 5 bits (0x00..0x1f) for PCI slot number (see definition of
+ * PCI_DEVFN) and we start at 0x18 (DEV_PCI_DEVICE), so there is hard upper
+ * limit on how many nodes can exist.
```

Comment:
FYI: There actually is a setting in AMD Family 10h/15h to remap those node devices to bus 0xfe and 0xff, giving a potential for up to 64 nodes: https://github.com/coreboot/coreboot/blob/4.11_branch/src/northbridge/amd/amdfam10/early_ht.c#L29